### ISSUE TITLE
Bug: dwarf axe gone from hand but other hand not free'd (after being paralyzed)

### ISSUE THREAD

**Issue by Pa9an** [Pyrdacor/Ambermoon.net#212]
After fighting with a mine spider and Egil being paralyzed (and being healed afterwards), the dwarf axe that he held before disappeared from the hand, but the X (due to the 2-handed weapon) is still in the other hand. Doing anything with that other hand leads to the game to crash.
The axe did not break during the fight. Unfortunately I overwrote the save game I had prior to the fight. But I attach the current one.
I'm on v1.1.0 but this can be reproduced also with a v1.0 test build.
![amb](https://user-images.githubusercontent.com/13840278/135877079-70b1ad31-7606-4502-b569-8ac12c5ab7b5.jpg)
[Save.10.zip](https://github.com/Pyrdacor/Ambermoon.net/files/7279114/Save.10.zip)

**Comment by Pa9an** [comment]
Apart from fixing the bug, do you think it is possible fixing the save game? :)

**Comment by Pyrdacor** [comment]
Sure just send it to me. You can just attach it here.

**Comment by Pa9an** [comment]
It's already in the first post ;)

**Comment by Pyrdacor** [comment]
Oh didn't see that. I will fix it tomorrow. You want the axe back or just the red cross gone?

**Comment by Pa9an** [comment]
Axe back would be nice, thanks!

**Comment by Pyrdacor** [comment]
Here you go.

[Save.10.zip](https://github.com/Pyrdacor/Ambermoon.net/files/7286551/Save.10.zip)

**Comment by Pyrdacor** [comment]
Most likely the same as #238

**Comment by komposter64** [comment]
I just ran into the same issue. Not sure if it is a coincidence, but it happend with the same weapon as with the op (dwarvern axe) on the same character, but Egil did not get paralyzed or otherwise impaird during the previous battles in my case. I also can't remeber that the weapon broke during that battle, allthough I just might have missed that happen (wouldn't be the first time). 

Additionally, after reloading my saved game, I noticed that Egil is now overburdened by 2kg, which he could not have been before, since I wouldn't be able to walk to where I saved (just outside Gemstone).

![grafik](https://user-images.githubusercontent.com/94021351/144729722-3cccbfd3-f3be-48fc-9365-4e813445fd9b.png)

**Comment by Pyrdacor** [comment]
Thanks for reporting. I guess the overweight stems from the missing strength boost of the broken axe.

Did you finish the fight? Was the broken axe available after the battle?

**Comment by komposter64** [comment]
Actually, I am not entirely sure anymore if I even had another combat after finding the Axe in one of the containers down in the cellar of Gemstone. I think there might have been at least one against those warriors that try to steal the dwarvern weapons, before leaving the cellar to explore the rest of Gemstone. I talk to the last of the dwarves, where I received the staff to open Donnar's mine and gave that to Egil as well, then left the city. I saved in front of the city gate (on the world map) and then returned to grandfather's house to store the loot and money I carried around after my visit to Gemstone and that's where I noticed the axe was gone. The game crashed with the error "key not found '0'" mentioned in #238, when I tried to replace it with another weapon. I loaded back both my crash save (inside grandfather's house) and then the previous save at Gemstone and am now unable to move away from there.

**Comment by Pyrdacor** [comment]
If you want, I can fix your savegame for you. Just attach it here and I will do so. Best you just add the Save.XX folder as a zip.

**Comment by komposter64** [comment]
Thanks, but that wont be necessary. I actually didn't lose that much progress and so I can check if I can reproduce the bug.

**Comment by Pyrdacor** [comment]
My investigation shows that the dwarf axe can't even break as it has a break chance of 0%. So it can't be due to breaking.

**Comment by Pyrdacor** [comment]
I tried several fights with dwarf axe againt mine spiders. I got cursed with several ailments etc but the axe always remained after the battle. :/

**Comment by komposter64** [comment]
I couldn't reproduce the bug either. Not knowing the code, would it be possible that it has something to do with the strength bonus the axe provides and being close to the characters carrying capacity?

**Comment by Pyrdacor** [comment]
Not really. Some event must remove the right hand weapon. But besides breaking there is none in battle and the dwarf axe can't break. I am not sure how this can happen.

**Comment by Pyrdacor** [comment]
By any chance, did you parry with the character when this happened?

**Comment by Pyrdacor** [comment]
When you parry an attack, instead of the armor, your weapon and shield can break. There I missed the code to remove two-handed weapons properly. But still the dwarf axe can not break so this should not happen anyway. :/

**Comment by Pa9an** [comment]
In my case, as I wrote, Egil was paralyzed. So he could neither attack nor parry. Yet, the axe was broken...or at least gone from the hand. Save game still attached to the OP.

**Comment by Pyrdacor** [comment]
Very hard to reproduce unfortunately. And the savegame afterwards does not help as I don't see the reason why the weapon has gone.

**Comment by cbrueffer** [comment]
My two cents from the bug report above (missed this one, sorry!): I was using a crowbar when the problem occurred very early in the game, so this is not unique to the dwarf axe. I cannot remember whether I was parrying since, as other reporters, I only noticed the problem a bit later, but I certainly cannot exclude the possibility. I've tried to reproduce the issue with an earlier save, but no luck so far.

Given that the handling around parrying is fixed in the upcoming release, let's see whether there are more reports after it has come out.

That said, thanks a lot for this project; I'd been meaning to re-play this game since I first played it back in Amiga days!
